### PR TITLE
Remove -swift-version 3 from a handful of tests and update them appro…

### PR DESCRIPTION
**Pull request by rudkx**
…priately.

These are all tests that would otherwise fail if the expression type
checker support for Swift 3 is removed.

I've moved some of the code from deleted Migrator tests into new
Constraints tests that verify that we do not support the constructs.

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Please smoke test

File: test/IDE/complete_func_reference.swift [hunk 1]
```diff
@@ -153,25 +151,13 @@ do {
 }
 
 // INT_ANY: Begin completions
-// INT_ANY-DAG: Decl{{.*}}/TypeRelation[Convertible]: intToAny(a:);
```
Comment (slavapestov) [inline test/IDE/complete_func_reference.swift]: Why did these completions go away? Is it a bug?

File: validation-test/compiler_crashers_fixed/26813-generic-enum-tuple-optional-payload.swift [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // This source file is part of the Swift.org open source project
 //
-// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
```
Comment (slavapestov) [inline validation-test/compiler_crashers_fixed/26813-generic-enum-tuple-optional-payload.swift]: Is this really necessary?

File: test/IDE/complete_func_reference.swift (hunk 1 above)
Comment (rudkx) [reply test/IDE/complete_func_reference.swift]: I asked that same question of @benlangmuir on the other PR. I haven't had a chance to look into it at all, but if it's unexpected we can open a JIRA and look into it.

File: validation-test/compiler_crashers_fixed/26813-generic-enum-tuple-optional-payload.swift (hunk 2 above)
Comment (rudkx) [reply validation-test/compiler_crashers_fixed/26813-generic-enum-tuple-optional-payload.swift]: Yes. Copyrights should be updated when files are updated.

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Comment by rudkx** [comment]
@benlangmuir I've updated the code completion results based on the current output. It turns out some completions didn't go away, they just had different content and different order. Apparently that's what we've been generating for a while now for -swift-version 4, but from glancing at the differences I have no idea if that would be expected. Should I open a JIRA for those differences, or do you already know why this is different?

File: test/IDE/complete_func_reference.swift (hunk 1 above)
Comment (slavapestov) [reply test/IDE/complete_func_reference.swift]: I filed a radar about it already but I thought you determined the change was correct. Until we find out can you leave this test as is for now?

Comment (rudkx) [reply test/IDE/complete_func_reference.swift]: @slavapestov I'm not planning on merging anything until @benlangmuir has a chance to reply to my question below.

**Comment by rudkx** [comment]
We have a bug tracking this yet unexplained difference. I discussed with @nathawes to make sure this seems okay to merge given that this is not a new behavior (i.e. -swift-version 4 already produced these completion results) and we have a bug tracking investigation of the differences.

**Pull request merged**